docs/code/SchemaComprehension.md

```diff
@@ -0,0 +1,210 @@
+# Schema comprehension in ML.NET
+
+This document describes in detail the under-the-hood mechanism that ML.NET uses to automate the creation of `IDataView` schema, with the goal to make it as convenient to the end user as possible, while not incurring extra computational costs.
+
+## Introduction
+
+Every dataset in ML.NET is an `IDataView`, which is, for the purposes of this document, a collection of rows that share the same columns. The set of columns, their names, types and other metadata is known as the *schema* of the `IDataView`, and it's represented as an `ISchema` object.
```

Comment:
>*schema*  [](start = 212, length = 9)

link to the schema section of the IDV Design Principles

Reply:
Added one above

---
In reply to: [204820706](https://github.com/dotnet/machinelearning/pull/572#discussion_r204820706) [](ancestors = 204820706)